src/SharpCompress/Compressors/DART/RLEStream.cs

```diff
@@ -0,0 +1,235 @@
+
+        /// <summary>
+        /// Position in decompressed data
+        /// </summary>
+        private long position;
+
+        /// <summary>
+        /// Buffer with currently used chunk of decompressed data
+        /// </summary>
+        private byte[] cmpBuffer;
+
+        /// <summary>
+        /// Position in buffer of decompressed data
+        /// </summary>
+        private int bufferPosition;
+
+        /// <summary>
+        /// Initializates a stream that decompresses ADC data on the fly
+        /// </summary>
+        /// <param name="stream">Stream that contains the compressed data</param>
+        /// <param name="compressionMode">Must be set to <see cref="CompressionMode.Decompress"/> because compression is not implemented</param>
+        public RLEStream(Stream stream, CompressionMode compressionMode = CompressionMode.Decompress)
+        {
+            if (compressionMode == CompressionMode.Compress)
+            {
+                throw new NotSupportedException();
+            }
+
+            this.stream = stream;
+        }
+
+        public override bool CanRead
+        {
+            get
+            {
+                return true;
+            }
+        }
+
+        public override bool CanSeek
+        {
+            get
+            {
+                return false;
+            }
+        }
+
+        public override bool CanWrite
+        {
+            get
+            {
+                return false;
+            }
+        }
+
+        public override long Length
+        {
+            get
+            {
+                throw new NotSupportedException();
+            }
+        }
+
+        public override long Position
+        {
+            get
+            {
+                return position;
+            }
+
+            set
+            {
+                throw new NotSupportedException();
+            }
+        }
+
+        public override void Flush()
+        {
+        }
+
+        protected override void Dispose(bool disposing)
+        {
+            if (isDisposed)
+            {
+                return;
+            }
+            isDisposed = true;
+            base.Dispose(disposing);
+        }
+
+        public override int Read(byte[] buffer, int offset, int count)
+        {
+            if (count == 0)
+            {
+                return 0;
+            }
+            if (buffer == null)
+            {
+                throw new ArgumentNullException(nameof(buffer));
+            }
+            if (count < 0)
+            {
+                throw new ArgumentOutOfRangeException(nameof(count));
+            }
+            if (offset < buffer.GetLowerBound(0))
+            {
+                throw new ArgumentOutOfRangeException(nameof(offset));
+            }
+            if ((offset + count) > buffer.GetLength(0))
+            {
+                throw new ArgumentOutOfRangeException(nameof(count));
+            }
+
+            int copied = 0;
+
+        fillBuffer:
```

Comment:
Mayyyyyyybe, I need to test.

And while most C# people tend to ignore gotos, they're used extensively in .NET frameworks and the IL optimizes them easy.

But if you prefer no goto I'll try to remove it, don't worry.